### DISCUSSION TITLE
Recurring Dates and Durations

### DISCUSSION THREAD

**Discussion by underdog57** [kuzmoyev/google-calendar-simple-api#128] [Q&A]
First off, great bit of code and as advertised, much easier to use. My questions relate to how duration:dates and duration:periods are handled. Using the data below, the dates and start times are correct, however the end times for days 2 and  3 are at 20:30 (10 hour delta) - which happens to be the same delta as day 1, rather than the 2 hour expected delta.

I've also tried a start datetime, Duration where duration is a timedelta, which generates an error : duration required, not timedelta.
(all objects are datetime wth timezone)
All event start/end are datetime object with timezone
Day 1:
start 2022-03-14 08:00:00-04:00
end 2022-03-14 18:00:00-04:00

Day 2: 
recurrence= Recurrence:periods((2022-03-23 10:30:00-04:00, 2022-03-23 12:30:00-04:00), (2022-03-14 08:00:00-04:00, 2022-03-14 18:00:00-04:00))

Thoughts? (and thanks)

**Comment by kuzmoyev** [comment]
Hello,

Sorry, I don't really follow what you are attempting to do and what doesn't work for you (or what you don't understand). Could you please share a piece of code that you are trying?

**Reply by underdog57** [reply]
Thanks for fast response. Here's the code:
Example 1 - all use America/New_York as timezone. The result is a correct "America" start time  and duration for the first meeting, but incorrect start and duration for subsequent meetings. Day2/3 end up with a start time of 10:30AM (UTC)
Example 2 - nextb_start/end astimezone(UTC) yields correct 10:30 a.m. America/New_York start time but with an end time of 20:00.

```
UTC = ZoneInfo('UTC')
tz = ZoneInfo('America/New_York')
start = datetime(2022,3,14,8,30, tzinfo=tz)
end =  datetime(2022,3,14,18,0, tzinfo=tz)
nexta_start = datetime(2022,3,17,10,30, tzinfo=tz)
nexta_end = datetime(2022,3,17,12,0, tzinfo=tz)
nextb_start = datetime(2022,3,18,10,30, tzinfo=tz)
nextb_end = datetime(2022,3,18,12,0, tzinfo=tz)
nextb_start = nextb_start.astimezone(UTC)
nextb_end = nextb_end.astimezone(UTC)

event = Event("Aarvark Meetings",
              start = start,
              end = end,
              recurrence = Recurrence.periods([(nexta_start, nexta_end),(nextb_start, nextb_end)])
              )
calendar.add_event(event)
```
In effect what I'm trying to do is generate 1 invitation (invite.ics) to cover 3 dates that are all part of the same event. Usually the days will be contiguous with identical start/end times. This works fine. Periods seems to be the way to handle non-contiguous dates except for the duration issue. I can't use RRule/Exdate as the result is an identical duration for all dates.

I've tried naive dates and added a timezone =  when creating the event. It makes no difference (as expected).

**Comment by kuzmoyev** [comment]
First of all, switching between timezones doesn't make a difference. Why do you do that?
Second, I still don't quite understand what DOESN'T work for you? What do you expect and what do you get instead?

I don't understand what is "Example 1" and what is "Example 2". You only create one event.
